compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/util/CompilerUtils.java

```diff
@@ -69,4 +77,17 @@ public static String getPackageIDStringWithMajorVersion(PackageID packageID) {
         return org + packageID.name + Names.VERSION_SEPARATOR.value + getMajorVersion(packageID.version.value);
     }
 
+    public static boolean isAssignmentToOptionalField(BLangAssignment assignNode) {
+        BLangNode varRef = assignNode.varRef;
+        if (varRef.getKind() != NodeKind.FIELD_BASED_ACCESS_EXPR) {
+            return false;
+        }
+        BLangFieldBasedAccess fieldAccessNode = (BLangFieldBasedAccess) varRef;
+        BType targetType = Types.getImpliedType(fieldAccessNode.expr.getBType());
+        if (targetType.tag != TypeTags.RECORD) {
+            return false;
+        }
+        BField field = ((BRecordType) targetType).fields.get(fieldAccessNode.field.value);
+        return field != null && Symbols.isOptional(field.symbol);
```

Comment:
Added this because I got the field as null when trying to build the Ballerina langlib. Will investigate and create a issue for that.

Reply:
Can we get rid of this now?

Reply:
It seems a new regression has been introduced since  #41998, while trying to desugar [this](https://github.com/ballerina-platform/ballerina-lang/blob/c4cd054f87796e5532de8c30be329e4e07c5bb37/misc/testerina/modules/testerina-core/src/main/ballerina/concurrentExecuter.bal#L62) query expression.

Reply:
Create #42479